File: java/base_n/Base16.java
package base_n;

/**
 * Hexadecimal base converter
 * @author	jcoder
 */
public class Base16 extends BaseNConverter
{
    public static final String ALPHABET = "0123456789ABCDEF";

    public Base16() {
        super(ALPHABET);
    }

    protected int ResolveCharToIndex(char c)
    {
        c = Character.toUpperCase(c);
        return super.resolveCharToIndex(c);
    }

}

File: csharp/Base16.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace BaseN
{
    /// <summary>
    /// Hexadecimal base converter
    /// </summary>
    public class Base16 : BaseNConverter
    {
        public const string ALPHABET = "0123456789ABCDEF";

        public Base16()
            : base(ALPHABET)
        {
        }

        protected override int ResolveCharToIndex(char c)
        {
            c = Char.ToUpper(c);
            return base.ResolveCharToIndex(c);
        }

    }
}
